contracts/test/integration/MultiSubnet.t.sol

```diff
@@ -0,0 +1,684 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.19;
+
+import "forge-std/Test.sol";
+import "../../src/errors/IPCErrors.sol";
+import {EMPTY_BYTES, METHOD_SEND} from "../../src/constants/Constants.sol";
+import {IpcEnvelope, BottomUpMsgBatch, BottomUpCheckpoint, ParentFinality, IpcMsgKind} from "../../src/structs/CrossNet.sol";
+import {FvmAddress} from "../../src/structs/FvmAddress.sol";
+import {SubnetID, Subnet, IPCAddress, Validator} from "../../src/structs/Subnet.sol";
+import {SubnetIDHelper} from "../../src/lib/SubnetIDHelper.sol";
+import {FvmAddressHelper} from "../../src/lib/FvmAddressHelper.sol";
+import {CrossMsgHelper} from "../../src/lib/CrossMsgHelper.sol";
+import {GatewayDiamond, FEATURE_MULTILEVEL_CROSSMSG} from "../../src/GatewayDiamond.sol";
+import {SubnetActorDiamond} from "../../src/SubnetActorDiamond.sol";
+import {SubnetActorGetterFacet} from "../../src/subnet/SubnetActorGetterFacet.sol";
+import {SubnetActorManagerFacet} from "../../src/subnet/SubnetActorManagerFacet.sol";
+import {SubnetActorCheckpointingFacet} from "../../src/subnet/SubnetActorCheckpointingFacet.sol";
+import {GatewayGetterFacet} from "../../src/gateway/GatewayGetterFacet.sol";
+import {GatewayManagerFacet} from "../../src/gateway/GatewayManagerFacet.sol";
+import {LibGateway} from "../../src/lib/LibGateway.sol";
+import {TopDownFinalityFacet} from "../../src/gateway/router/TopDownFinalityFacet.sol";
+import {CheckpointingFacet} from "../../src/gateway/router/CheckpointingFacet.sol";
+import {XnetMessagingFacet} from "../../src/gateway/router/XnetMessagingFacet.sol";
+import {DiamondCutFacet} from "../../src/diamond/DiamondCutFacet.sol";
+import {GatewayMessengerFacet} from "../../src/gateway/GatewayMessengerFacet.sol";
+import {DiamondLoupeFacet} from "../../src/diamond/DiamondLoupeFacet.sol";
+import {DiamondCutFacet} from "../../src/diamond/DiamondCutFacet.sol";
+import {IntegrationTestBase} from "../IntegrationTestBase.sol";
+import {L2GatewayActorDiamond, L1GatewayActorDiamond} from "../IntegrationTestPresets.sol";
+import {TestUtils, MockIpcContract, MockIpcContractPayable, MockIpcContractFallback} from "../helpers/TestUtils.sol";
+import {FilAddress} from "fevmate/utils/FilAddress.sol";
+import {MerkleTreeHelper} from "../helpers/MerkleTreeHelper.sol";
+
+import {IERC20} from "openzeppelin-contracts/token/ERC20/IERC20.sol";
+import {ERC20PresetFixedSupply} from "../helpers/ERC20PresetFixedSupply.sol";
+import {IERC20Errors} from "openzeppelin-contracts/interfaces/draft-IERC6093.sol";
+
+import "forge-std/console.sol";
+
+contract MultiSubnetTest is Test, IntegrationTestBase {
+    using SubnetIDHelper for SubnetID;
+    using CrossMsgHelper for IpcEnvelope;
+
+    GatewayDiamond public rootGateway;
+    GatewayGetterFacet public rootGatewayGetter;
+    GatewayManagerFacet public rootGatewayManager;
```

Comment:
With the sugar I proposed above, we can just have the rootGateway here, and methods can do: rootGateway.getter()... Much nicer!